### DISCUSSION TITLE
Ask for password if saving an edited file in root directory

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#4020] [Ideas]
I know this is not going to get added / also not something I personally would use all the time but just something I noticed in different programs that I wanted to suggest.
Some editors have the ability to be able to open up root based text files and if any modifications are made a dialog box appears asking for user to enter their password to approve the changes.  I assume they are using something like "pkexec" but I am not sure.
Again not something I would use often, but found it convenient at times.
Not sure if any other users would find this helpful.

Thank You

**Comment by elextr** [comment]
A plugin that copies the root file to a normal copy until it is edited and saved by Geany at which point the plugin can find the root permissions (and password or whatever) and then the root file replaced by the edited copy.

Such a plugin is not part of Geany.

But "somebody" could produce such a plugin and contribute it to Geany-plugins collection if the request is on that repository, not Geany.

**Comment by advice2020** [comment]
I created a post for this in the plugins section which can be found here if anyone is interested
https://github.com/geany/geany-plugins/issues/1378

**Comment by b4n** [comment]
IMO this should ideally use GIO's `admin://` scheme, which handles all this. Of course this requires Geany to use GIO for this, which is only partially the case now.

A plugin could work indeed, but that kind of sounds like duplicating a potentially tricky feature.
